cmd/kubeadm/app/componentconfigs/kubelet.go

```diff
@@ -231,3 +239,18 @@ func isServiceActive(name string) (bool, error) {
 	}
 	return initSystem.ServiceIsActive(name), nil
 }
+
+// MutateCgroupDriver can be called to set the KubeletConfiguration cgroup driver to systemd
+// Currently this cannot be as part of Default() because the function is called for
+// upgrades too, which can break existing nodes after a kubelet restart.
```

Comment:
Do we plan to move this back to default in next releases or do we think that a different upgrade path will be always required given that we are seeing users upgrading very old clusters?
Also, for sake of future memory, what about adding a note on Default about this field being defaulted out of band?

Reply:
as i've mentioned, the defaulting being the same for all commands is problematic.
we have a similar case for the "docker cgroup driver detection" code, that i like to move to dockershim.

since i really don't like keeping `MutateCgroupDriver()` forever i wonder what is best here...

i don't wish to do add this change in Default() for this release because it would break upgrading users in just one release, but we could say that in 1 more release kubeadm will start doing that in all of its commands, which means we are going to merge the code in Default() and users have 1 release to adapt their nodes / CRs (move to the systemd driver).

i like this approach better, because this code separation is not ideal.

if you agree to that i can add a comment over Default() about MutateCgroupDriver().
..and update the release note too.

Reply:
Given that this could be potentially disruptive, I think we should:
- advertise the change through usual channels
- have a warning in upgrade plan for at least one cycle (you are running with two cgroup drivers, see the doc to migrate)
- (finally) make this our default

Reply:
> advertise the change through usual channels

i can send a notification to k-dev and #kubeadm

> have a warning in upgrade plan for at least one cycle (you are running with two cgroup drivers, see the doc to migrate)

i can see where in `upgrade plan` this can be added for this PR.
the migration docs are waiting for review and i will like to them from the release note too.